### ISSUE TITLE
Java 22 breaks typing in console on windows.

### ISSUE THREAD

**Issue by andrewkm** [PaperSpigot/Paper#10405]
### Expected behavior

Smooth console input via Windows Server 2022 cmd.exe without glitches.

### Observed/Actual behavior

Glitchy buggy console input.
Freezing, etc.

### Steps/models to reproduce

Java 22; typing in consoles is **broken** on windows with paper, works on Java 21.
Java 22; typing in consoles **works fine** on windows with bungeecord/waterfall.

Testing on fresh Paper install with no plugins for sanity, same result, borked on 22 good on 21.
Basically it's all glitched, freezes, random keys don't register, etc.

### Plugin and Datapack List

Unrelated, tested on fresh install.

### Paper version

`This server is running Paper version git-Paper-478 (MC: 1.20.4) (Implementing API version 1.20.4-R0.1-SNAPSHOT) (Git: 5436d44)`

**Comment by electronicboy** [comment]
I don't have x86 hardware worth firing up for this, so, span up the windows VM I have sitting around on this arm laptop;

1) Terminal input inside of windows terminal generally seems kinda janky, using conhost provides a much better experience in general
2) counter to 1, it seemed to have gotten stuck less, but, it still ends up getting locked up, it's almost as if the terminal stuff is just not reading the input for some reason, if you keep spamming it eventually seems to pick back up again.

I don't have the desire/will to dig much deeper into this with the lack of a real setup here especially as it's working fine for the environments I can support

**Comment by andrewkm** [comment]
Just an update, this bug persists on JAVA 22.0.1

**Comment by Malfrador** [comment]
This seems to be a Powershell 5 specific issue. The console is working fine both with the good old cmd.exe, and it also works without any issues with Powershell 7.

Seems to be caused by this [JDK change](https://bugs.openjdk.org/browse/JDK-8308591). As a workaround, starting the server with `-Djdk.console=java.base` to revert to the old JDK behaviour fixes the issue. I would recommend just using PS7 though.

**Comment by electronicboy** [comment]
I thought that I got somebody to create an issue report but I'm blind, that is likely related to the crash that people get when double clicking on the server jar in order to run it

**Comment by andrewkm** [comment]
The `-Djdk.console=java.base` workaround does work on **JAVA 22.0.1**. Thanks for that!

**Comment by MinecraftMan1013number2** [comment]
I am also having this issue and I have found that the issue does NOT happen with version paper-1.20-17, but DOES happen in version paper-1.20.1-196, all with Java 22. I can try and find the exact paper version where the issue starts, but It'll take some time.

**Comment by jgibbz328** [comment]
I have updated to Powershell 7, but still having the same issue.  I am a newb and not sure where to put the 
-Djdk.console=java.base  (i tried putting it in the start.bat file but that didn't work)

**Comment by andrewkm** [comment]
> This seems to be a Powershell 5 specific issue. The console is working fine with the good old cmd.exe, and it also works without any issues with Powershell 7.
> 
> 
> 
> Seems to be caused by this [JDK change](https://bugs.openjdk.org/browse/JDK-8308591). As a workaround, starting the server with `-Djdk.console=java.base` to revert to the old JDK behaviour fixes the issue. I would recommend just using PS7 though, its much nicer anyways.

For the record I'm not using powershell. I am in fact using good old cmd.exe

**Comment by andrewkm** [comment]
As an update, this is still broken via:

Paper 1.21 Build 101
Windows Server 2022 **cmd.exe**
Java 22.0.2
